### PR TITLE
fix(client): inject bootstrap tags to the top of head

### PR DESCRIPTION
**Pull request by ahmaxed**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Closes #XXXXX

### CODE REVIEW

**Review by moT01 — APPROVED** [review]
LGTM 🎉

File: client/gatsby-ssr.js [hunk 1]
```diff
@@ -32,3 +32,19 @@ export const onRenderBody = ({
...
+
+export const onPreRenderHTML = ({
+  getHeadComponents,
+  replaceHeadComponents
+}) => {
+  const headComponents = getHeadComponents();
+  headComponents.sort((x, y) => {
+    if (x.key === 'boostrap-min-preload' || x.key === 'boostrap-min') {
+      return -1;
+    } else if (y.key === 'TypographyStyle' || x.key === 'boostrap-min') {
```
Comment (ShaunSHamilton) [inline client/gatsby-ssr.js:44]:
```suggestion
    } else if (y.key === 'TypographyStyle') {
```

I do not think this is ever reached

Comment (ShaunSHamilton) [reply client/gatsby-ssr.js:44]:
```suggestion
    } else if (y.key === 'TypographyStyle' || y.key === 'boostrap-min') {
```
Actually, I am wondering if it is supposed to be the above?

Comment (ahmadabdolsaheb) [reply client/gatsby-ssr.js]: I added the TypographyStyle by mistake.

Comment (ojeytonwilliams) [inline client/gatsby-ssr.js]:
```suggestion
    } else if (y.key === 'bootstrap-min-preload' || y.key === 'boostrap-min') {
```
I think the logic here is 
> if (x is bootstrap) then x comes first
> if (y is bootstrap) then y comes first

and we don't care about the order other than the two bootstraps should be first.

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
Nice catch, Ahmad.  I glanced at the function, but didn't look closely enough.

File: client/gatsby-ssr.js [hunk 2]
```diff
@@ -32,3 +32,19 @@ export const onRenderBody = ({
...
+
+export const onPreRenderHTML = ({
+  getHeadComponents,
+  replaceHeadComponents
+}) => {
+  const headComponents = getHeadComponents();
+  headComponents.sort((x, y) => {
+    if (x.key === 'boostrap-min-preload' || x.key === 'boostrap-min') {
+      return -1;
+    } else if (y.key === 'boostrap-min-preload' || y.key === 'boostrap-min') {
```
Comment (ShaunSHamilton) [inline client/gatsby-ssr.js:44]:
```suggestion
    if (x.key === 'bootstrap-min-preload' || x.key === 'bootstrap-min') {
      return -1;
    } else if (y.key === 'bootstrap-min-preload' || y.key === 'bootstrap-min') {
```

**Review by ShaunSHamilton — DISMISSED** [review]

**Review by ojeytonwilliams — APPROVED** [review]

**Comment by ojeytonwilliams** [comment]
Looks like we all agree on the compare function :grin:

**Comment by nhcarrigan** [comment]
Building this locally does appear to resolve our CSS issue (`npm run build`, not `npm run develop`) but I see the following errors during the client build:

image

File: client/gatsby-ssr.js (hunk 2 above)
Comment (ahmadabdolsaheb) [reply client/gatsby-ssr.js]: Good catch.  The original keys were misspelled. I updated all of the accordingly.

**Comment by ojeytonwilliams** [comment]
@nhcarrigan those errors have been around for a while:
https://dev.azure.com/freeCodeCamp-org/freeCodeCamp/_build/results?buildId=648&view=logs&j=275f1d19-1bd8-5591-b06b-07d489ea915a&t=29b11093-f299-52c9-fc3e-b4e81abf4ac1&l=205

so I think they're unrelated.  They could use fixing, though, and making sure that that doesn't break the styles.

File: client/gatsby-ssr.js (hunk 1 above)
Comment (ahmadabdolsaheb) [reply client/gatsby-ssr.js]: this should be resolved.

**Review by ShaunSHamilton — APPROVED** [review]
This LGTM 👍

**Pull request merged**